src/components/date.rs

```diff
@@ -21,9 +21,40 @@ use std::str::FromStr;
 
 use super::{
     duration::{normalized::NormalizedDurationRecord, TimeDuration},
-    MonthDay, Time, YearMonth,
+    MonthCode, MonthDay, Time, YearMonth,
 };
 
+// TODO: PrepareTemporalFields expects a type error to be thrown when all partial fields are None/undefined.
+/// A partial Date that may or may not be complete.
+#[derive(Debug, Clone, Copy)]
+pub struct PartialDate {
+    pub(crate) year: Option<i32>,
+    pub(crate) month: Option<i32>,
+    pub(crate) month_code: Option<MonthCode>,
+    pub(crate) day: Option<i32>,
+}
```

Comment:
What I said in the Temporal meeting about the requirement being "day AND (month OR monthCode) AND (year OR (era AND eraYear))" reminded me that this should probably have optional fields for `era` and `era_year`.

Reply:
Oh, good point. I had meant to add those in one the update and forgot too.